Stripe/STPBankSelectionTableViewCell.m

```diff
@@ -0,0 +1,115 @@
+//
+//  STPBankSelectionTableViewCell.m
+//  Stripe
+//
+//  Created by David Estes on 8/9/19.
+//  Copyright © 2019 Stripe, Inc. All rights reserved.
+//
+
+#import "STPBankSelectionTableViewCell.h"
+
+#import "STPApplePayPaymentOption.h"
+#import "STPCard.h"
+#import "STPImageLibrary+Private.h"
+#import "STPLocalizationUtils.h"
+#import "STPPaymentMethod.h"
+#import "STPPaymentMethodCard.h"
+#import "STPPaymentOption.h"
+#import "STPSource.h"
+#import "STPTheme.h"
+
+@interface STPBankSelectionTableViewCell ()
+
+@property (nonatomic, readwrite) STPFPXBankBrand bank;
+@property (nonatomic, strong, readwrite) STPTheme *theme;
+
+@property (nonatomic, strong, readwrite) UIImageView *leftIcon;
+@property (nonatomic, strong, readwrite) UILabel *titleLabel;
+@property (nonatomic, strong, readwrite) UIActivityIndicatorView *activityIndicator;
+
+@end
+
+@implementation STPBankSelectionTableViewCell
+
+- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(nullable NSString *)reuseIdentifier {
+    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
+    if (self) {
+        // Left icon
+        UIImageView *leftIcon = [[UIImageView alloc] init];
+        _leftIcon = leftIcon;
+        [self.contentView addSubview:leftIcon];
+
+        // Title label
+        UILabel *titleLabel = [UILabel new];
+        _titleLabel = titleLabel;
+        [self.contentView addSubview:titleLabel];
+
+        // Loading indicator
+        UIActivityIndicatorView *activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+        _activityIndicator = activityIndicator;
+        _activityIndicator.hidesWhenStopped = YES;
+        [self.contentView addSubview:activityIndicator];
+    }
+    return self;
+}
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+
+    CGFloat midY = CGRectGetMidY(self.bounds);
+    CGFloat padding = 15.0;
+    CGFloat iconWidth = 26.0;
+
+    // Left icon
+    [self.leftIcon sizeToFit];
+    self.leftIcon.center = CGPointMake(padding + (iconWidth / 2.0f), midY);
+
+    // Activity indicator
+    self.activityIndicator.center = CGPointMake(CGRectGetWidth(self.bounds) - padding - CGRectGetMidX(self.activityIndicator.bounds), midY);
+
+    // Title label
+    CGRect labelFrame = self.bounds;
+    // not every icon is `iconWidth` wide, but give them all the same amount of space:
+    labelFrame.origin.x = padding + iconWidth + padding;
+    labelFrame.size.width = CGRectGetMinX(self.activityIndicator.frame) - padding - labelFrame.origin.x;
+    self.titleLabel.frame = labelFrame;
```

Comment:
why not autolayout?

Reply:
This is a copypasta from our existing TableViewControllers. I don't think we use autolayout anywhere in the SDK.

Reply:
😢